app/Admin/src/Http/Requests/Auth/LoginRequest.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace App\Admin\Http\Requests\Auth;
+
+use App\Platform\Enums\RoleName;
+use Illuminate\Auth\Events\Lockout;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\RateLimiter;
+use Illuminate\Support\Str;
+use Illuminate\Validation\Rule;
+use Illuminate\Validation\ValidationException;
+
+class LoginRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     */
+    public function authorize(): bool
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
+     */
+    public function rules(): array
+    {
+        return [
+            'email' => ['required', Rule::exists('users', 'email'), 'email', 'ends_with:learnhub.mk'],
+            'password' => ['required']
+        ];
+    }
+
+    /**
+     * Get the login credentials from the request.
+     *
+     * @return array<string, mixed>
+     */
+    public function credentials(): array
+    {
+        $field = filter_var($this->input('email'), FILTER_VALIDATE_EMAIL);
+
+        return [
+            'email' => $field,
+            'password' => $this->input('password'),
+        ];
+    }
+
+    /**
+     * Attempt to authenticate the request's credentials.
+     *
+     * @throws \Illuminate\Validation\ValidationException
+     */
+    public function authenticate(): void
```

Comment:
This method triggers the authentication logic so it should probably be in the controller or a service class rather than the request class.

Same goes for the method below.

Also, see if you can use a middleware to clean up the rate limiting checks.